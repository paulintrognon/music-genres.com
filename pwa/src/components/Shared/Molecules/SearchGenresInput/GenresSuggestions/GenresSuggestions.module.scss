.suggestion {
  display: block;
  width: 100%;
  padding: 0.75rem 1rem;
  background-color: var(--color-secondary);
  
  a {
    text-decoration: underline;
  }
}

.hover:hover {
  background-color: var(--color-secondary-light);
  box-shadow: 0 1px 10px 0 rgba(0,0,0,.19);
  position: relative;
}