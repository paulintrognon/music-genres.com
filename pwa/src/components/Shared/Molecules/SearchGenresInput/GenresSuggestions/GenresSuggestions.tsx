import Link from 'next/link'
import { MusicGenreBaseType } from '../../../../../types/MusicGenre/MusicGenreBaseType'
import classnames from 'classnames'
import classes from './GenresSuggestions.module.scss'
import Loader from '../../../Atoms/Loader/Loader'

type Props = {
  isLoading: boolean
  genreName: string
  suggestions: MusicGenreBaseType[]
  slugToHref: (slug: string) => string
}
const GenresSuggestions: React.FC<Props> = ({ isLoading, genreName, suggestions, slugToHref }) => {
  if (isLoading) {
    return (
      <div className={classes.suggestion} style={{ textAlign: 'center' }}>
        <Loader />
      </div>
    )
  }
  if (suggestions.length === 0) {
    return (
      <div className={classes.suggestion}>
        &quot; <i>{genreName}</i> &quot; does not exist yet.{' '}
        <Link href={`/add/video/${genreName}`}>
          <a>Create it!</a>
        </Link>
      </div>
    )
  }
  return (
    <div>
      {suggestions.map((suggestion) => (
        <Link key={suggestion.slug} href={slugToHref(suggestion.slug)}>
          <a className={classnames(classes.suggestion, classes.hover)}>{suggestion.name}</a>
        </Link>
      ))}
    </div>
  )
}
export default GenresSuggestions
