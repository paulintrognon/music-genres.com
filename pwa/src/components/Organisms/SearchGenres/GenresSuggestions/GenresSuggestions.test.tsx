import React from 'react'
import createRenderer from 'react-test-renderer/shallow'
import GenresSuggestions from './GenresSuggestions'

const renderer = createRenderer.createRenderer()
const slugToHref = (slug: string): string => `/path/to/${slug}`
const genreName = 'Super Genre'
const suggestions = [
  { id: 1, name: 'Genre 1', slug: 'genre-1' },
  { id: 2, name: 'Genre 2', slug: 'genre-2' },
  { id: 3, name: 'Genre 3', slug: 'genre-3' },
]

it('should render the GenresSuggestions', () => {
  expect(
    renderer.render(
      <GenresSuggestions
        isLoading={false}
        genreName={genreName}
        suggestions={suggestions}
        slugToHref={slugToHref}
      />
    )
  ).toMatchSnapshot()
})

it('should render "loader" if loading', () => {
  expect(
    renderer.render(
      <GenresSuggestions
        isLoading={true}
        genreName={genreName}
        suggestions={suggestions}
        slugToHref={slugToHref}
      />
    )
  ).toMatchSnapshot()
})

it('should render "no result" if no suggestions', () => {
  expect(
    renderer.render(
      <GenresSuggestions
        isLoading={false}
        genreName={genreName}
        suggestions={[]}
        slugToHref={slugToHref}
      />
    )
  ).toMatchSnapshot()
})
