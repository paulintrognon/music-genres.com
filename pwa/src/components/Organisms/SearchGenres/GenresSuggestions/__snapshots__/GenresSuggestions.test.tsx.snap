// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render "loader" if loading 1`] = `
<div
  className="suggestion"
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <Loader />
</div>
`;

exports[`should render "no result" if no suggestions 1`] = `
<div
  className="suggestion"
>
  " 
  <i>
    Super Genre
  </i>
   " does not exist yet.
   
  <Link
    href="/add/video/Super Genre"
  >
    <a>
      Create it!
    </a>
  </Link>
</div>
`;

exports[`should render the GenresSuggestions 1`] = `
<div>
  <Link
    href="/path/to/genre-1"
  >
    <a
      className="suggestion hover"
    >
      Genre 1
    </a>
  </Link>
  <Link
    href="/path/to/genre-2"
  >
    <a
      className="suggestion hover"
    >
      Genre 2
    </a>
  </Link>
  <Link
    href="/path/to/genre-3"
  >
    <a
      className="suggestion hover"
    >
      Genre 3
    </a>
  </Link>
</div>
`;
